sdk/search/Azure.Search.Documents/src/Spatial/SpatialProxyFactory.cs

```diff
@@ -0,0 +1,109 @@
+                ["Microsoft.Spatial.GeographyPoint"] = value => new GeographyPointProxy(value),
+                ["Microsoft.Spatial.GeographyPolygon"] = value => new GeographyPolygonProxy(value),
+            };
+
+        /// <summary>
+        /// Gets a value indicating whether the given <paramref name="type"/> can be created by this factory.
+        /// </summary>
+        /// <param name="type">The type to check.</param>
+        /// <returns>A value indicating whether <paramref name="type"/> can be created by this factory.</returns>
+        public static bool CanCreate(Type type)
+        {
+            if (type is null)
+            {
+                return false;
+            }
+
+            return TryGetFactory(type, out _);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="GeographyProxy"/> from the given <paramref name="value"/> if supported.
+        /// <seealso cref="CanCreate(Type)"/>
+        /// </summary>
+        /// <param name="value">The value to proxy.</param>
+        /// <returns>A <see cref="GeographyProxy"/> from the given <paramref name="value"/> if supported, or null if <paramref name="value"/> is null.</returns>
+        /// <exception cref="NotSupportedException">The <paramref name="value"/> type is not supported.</exception>
+        public static GeographyProxy Create(object value)
+        {
+            if (value is null)
+            {
+                return null;
+            }
+
+            if (TryCreate(value, out GeographyProxy proxy))
+            {
+                return proxy;
+            }
+
+            throw new NotSupportedException($"Type {value.GetType()} is not supported");
+        }
+
+        /// <summary>
+        /// Attempts to creates a <see cref="GeographyProxy"/> from the given <paramref name="value"/> if supported.
+        /// <seealso cref="CanCreate(Type)"/>
+        /// </summary>
+        /// <param name="value">The value to proxy.</param>
+        /// <param name="proxy">The proxied value if supported.</param>
+        /// <returns>True if the <paramref name="value"/> could be proxied; otherwise, false.</returns>
+        internal static bool TryCreate(object value, out GeographyProxy proxy)
+        {
+            if (value is { })
+            {
+                Type type = value.GetType();
+                if (TryGetFactory(type, out Func<object, GeographyProxy> factory))
+                {
+                    proxy = factory(value);
+                    return true;
+                }
+            }
+
+            proxy = null;
+            return false;
+        }
+
+        private static bool TryGetFactory(Type type, out Func<object, GeographyProxy> factory)
+        {
+            // TODO: Should we compare the public key token as well, or is that too restrictive or unnecessarily slow?
+            AssemblyName assemblyName = type.Assembly.GetName();
+            if (string.Equals(assemblyName.Name, Name, StringComparison.OrdinalIgnoreCase))
```

Comment:
We should discuss that with folks during the review.  I'm okay going with anything that duck types into the shape of Microsoft.Spatial for the very, very few properties we access.

Reply:
For performance I'd rather just do assembly name + type name anyway. I'll remove the TODO comment.